### DISCUSSION TITLE
All set up and connected - How to setup buttons/switches

### DISCUSSION THREAD

**Discussion by Apok1** [Koenkk/zigbee2mqtt#16815] [Devices]
Hi,

In short - how the heck do I make pressing a button on my TS0043 do a thing, like turn a S26R2ZB to ON?!

A while ago I setup Home assistant on a pi3B+.  I used ZHA as it was there and got everything set up very easily, I could set pressing/double clicking any button to toggle,/turnon etc any other device by (using device/automations in HA).  However the TS0043 light switch battery dies after 3-4 weeks and is not very responsive either.

Under advise I have added a USB extension, removed ZHA completely and installed zigbee2mqtt + Mosquitto broker.  They work fine, all devices connected and showing in the HA devices.  I can add them to a HA dashboard and turn them on and off.  I love that I can see the LQI and have even discovered a dead-ish behind an old chimney  zone - great!

What I cannot do is set the button clicks to trigger the same actions as before.  Going to the device/automations in HA i only have  "3_single" action as a trigger under that device, not each button option as before (1_single, 1_double, 1_hold, 2_single, 2_double, 2_hold, 3_single, 3_double, 3_hold).

I see binding as an option under the zigbee2mqtt web ui, but this seems to relate to clusters (?) which I see mapped (copy below) - this doesn't seem to be what I want.

0x5c0272fffe2293e6
Endpoint 1
Output clusters
genOta
genTime
Input clusters
genBasic
genPowerCfg
genOnOff
Endpoint 2
Input clusters
genPowerCfg
genOnOff
Endpoint 3
Input clusters
genPowerCfg
genOnOff

I read the support device page and can see the actions I want, but no way to configure them.

I have read through the zigbee2mqtt guide again and found this;
 https://www.zigbee2mqtt.io/guide/usage/integrations/home_assistant.html#customizing-discovery

But that leads you off to this in HA
https://www.home-assistant.io/integrations/device_trigger.mqtt/

Which leads me off to this...
https://www.home-assistant.io/integrations/button.mqtt/

I can see there is a MQTT option in the HA/Device/Automations asking for 'topics' (?).  I found a page on topics, which seems to assume you already know all about them - https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html.

After all this I am no further to knowing what I should do and I'm feeling a bit like Alice, tumbling down the rabbit hole - am I zoning in on the solution off way off the beaten path? I see other posts about connections, seem you are writing the commands, do I need to do this if using HA? Where does it go and what is the syntax?

I would appreciate any assistance, I'm learning as I go here, but this is still all new to me and I feel like I might have missed something obvious.

Many Thanks,

**Comment by forsberg72** [comment]
I have just this same feeling. How to define button actions? I have a dimmer switch and two Ikea Troådfri on/off buttons. I would like to know how to define actions for buttons. I think there's no good documentation.

**Reply by Apok1** [reply]
Hi, my issue was the button options were not showing up.  not sure that is the same but to any one who wants it, my solution was:
1. Press the actual buttons you want to use, this somehow remind the device it has the buttons, I could then go to the device/automations in HA and set the automations.
2. One button would still not appear, i tried some things but it was only after going back the next day it suddenly showed - give it some time

Is your issue more about generally setting automations or that the specifc buttons do not show within the automations section?  If it's the latter [check this guy out](https://www.youtube.com/watch?v=KXTs5_x_T5c)

Hope that helps
